addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -621,7 +621,12 @@ odoo.define('website_slides.fullscreen', function (require) {
                 return self._renderSlide();
             }).then(function() {
                 if (slide._autoSetDone && !session.is_website_user) {  // no useless RPC call
```

Comment:
@tde-banana-odoo ,

can not avoid a second RPC call to set a completed slide in case of slide type is documentation, presentation?

If I am right, here logic is that if slide type is documentation or presentation, A slide should go in the completed state after viewed by the public user.

can not put this business logic/flow in a server-side instead of calling another RPC?

Reply:
Hello,

Looking a bit at the code, issue is that in fullscreen we load as much information as possible to keep in fullscreen and just change slide without loading anything from server (unless there is some html content to fetch). I think it would imply marking the whole channel as completed even before going through the slides.

Did you check a bit this flow ?

Thanks,